Expensify/App
# Check for unvalidated accounts and change in password flow

HOLD until the web-e and auth PRs go live i.e,
- https://github.com/Expensify/Auth/pull/5382 AND
- https://github.com/Expensify/Web-Expensify/pull/30507

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/156783

### Tests
1. On dev first go to ./script/sql.sh and run delete from notifications to begin on a clean slate.
2. Go to e.cash
3. Enter your Email. Confirm you see the forgot button,
![image](https://user-images.githubusercontent.com/14356145/111828074-fcf89e80-88a7-11eb-8ef8-0a78e1b57064.png)
4. Click it. For dev run `php /vagrant/Web-Expensify/script/notifyall.php` from within vssh.
5. Check the email. Confirm its magic link is of the format `<baseUrl>/setpassword/<code>`. Don't click it just yet.
6. Confirm the UI for the existing e.cash tab has updated to be,
![image](https://user-images.githubusercontent.com/14356145/111828303-582a9100-88a8-11eb-9e62-481b28b99ef8.png)
7. Open console. Click "Resend Link" and confirm you see the console log "Account forgot password: Sending reset password link."
8. Click the link from the email at step 5 and confirm you are redirected to set your password. Set your password and confirm you are logged in. Log out.
9. Now attempt to log into e.cash using an account that's not validated.
10. Confirm you are redirected to this UI
![image](https://user-images.githubusercontent.com/14356145/111828942-44335f00-88a9-11eb-80c5-779673dd13cd.png)
11. Open console and click "Resend link". Confirm you see the log "Account is unvalidated: Sending validation link."
12. For dev run `php /vagrant/Web-Expensify/script/notifyall.php` from within vssh. You might have to run `update notifications set json = json_set(json(notifications.json), '$.to', '<yourExpensifyEmail>');` if the unvalidated email does not belong to expensify since we only send emails out to expensify.
13. Check the email and confirm its link is of the format `<baseURL>/setpassword/<code>`. Click it and confirm you can set your password and you are logged in once done.



### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
